### PR TITLE
feat: Add a feature flag for S3 folder sync

### PR DESCRIPTION
**Pull request by jeethufoc**
Add a feature flag to enable/disable AWS S3 folder sync while creating QA/LQA users in webhook
Ticket - [LSI-1800](https://oat-sa.atlassian.net/browse/LSI-1800)

[LSI-1800]: https://oat-sa.atlassian.net/browse/LSI-1800?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/EventSubscriber/GeneratedUserIngestControllerSubscriber.php [hunk 1]
```diff
@@ -110,6 +117,10 @@ public function onLineItemUpdated(LineItemUpdated $event): void
             $groupResolver
         );
 
+        if (!$this->folderSyncEnabled) {
+            return;
+        }
```
Comment (SergiiTao) [inline src/EventSubscriber/GeneratedUserIngestControllerSubscriber.php]: early return is good when you want to interrupt a function when something goes wrong or doesn't met some conditions no matter what. but for skipping a single line (even though it's the last one :) I think the classic way is better
```
        if ($this->folderSyncEnabled) {
            $this->userFolderSync->sync($date);
        }
```

File: src/EventSubscriber/GeneratedUserIngestControllerSubscriber.php [hunk 2]
```diff
@@ -60,10 +65,12 @@ public function __construct(
         FolderSyncService $userFolderSync,
         CreateUserServiceContext $createUserServiceContext,
         string $groupPrefix,
-        bool $enabled
+        bool $enabled,
+        bool $folderSyncEnabled
```
Comment (SergiiTao) [inline src/EventSubscriber/GeneratedUserIngestControllerSubscriber.php]: I think the name `isFolderSyncEnabled` is more eloquent

**Review by SergiiTao — CHANGES_REQUESTED** [review]
good job! just 2 minor comments

File: tests/Unit/EventSubscriber/GeneratedUserIngestControllerSubscriberTest.php [hunk 3]
```diff
@@ -73,12 +73,67 @@ public function testOnLineItemUpdatedIgnoreOnDisabled(): void
...
+        $service = new GeneratedUserIngestControllerSubscriber(
+            $logerMock,
+            $createServiceMock,
+            $generateGroupIdsServiceMock,
+            $ltiInstanceRepositoryMock,
+            $lineItemRepositoryMock,
+            $userFolderSyncMock,
+            new CreateUserServiceContext(['test1'], ['tao1'], 10, true),
+            'test',
+            false,
```
Comment (mianchenia) [inline tests/Unit/EventSubscriber/GeneratedUserIngestControllerSubscriberTest.php]: How do we check that sync was ignored?

File: tests/Unit/EventSubscriber/GeneratedUserIngestControllerSubscriberTest.php [hunk 4]
```diff
@@ -73,12 +73,67 @@ public function testOnLineItemUpdatedIgnoreOnDisabled(): void
...
             false
         );
 
         $service->onLineItemUpdated(new LineItemUpdated(['test1']));
     }
 
+    public function testOnLineItemUpdatedIgnoreOnEmptyList(): void
```
Comment (mianchenia) [inline tests/Unit/EventSubscriber/GeneratedUserIngestControllerSubscriberTest.php]: This test shares same source code that `testOnLineItemUpdatedIgnoreFolderSyncOnDisabled ` does. 
I don't understand how do they test different stuff with the same source code.

**Review by mianchenia — CHANGES_REQUESTED** [review]

**Comment by mianchenia** [comment]
Good job! Please have a look on my questions.

**Review by mianchenia — APPROVED** [review]
LGTM!

**Review by SergiiTao — APPROVED** [review]

**Pull request merged**